pkg/scheduler/framework/plugins/nodevolumelimits/non_csi.go

```diff
@@ -267,17 +268,36 @@ func (pl *nonCSILimits) Filter(ctx context.Context, _ *framework.CycleState, pod
 	return nil
 }
 
-func (pl *nonCSILimits) filterVolumes(volumes []v1.Volume, namespace string, filteredVolumes sets.String) error {
+func (pl *nonCSILimits) filterVolumes(pod *v1.Pod, volumes []v1.Volume, namespace string, filteredVolumes sets.String) error {
 	for i := range volumes {
 		vol := &volumes[i]
 		if id, ok := pl.filter.FilterVolume(vol); ok {
 			filteredVolumes.Insert(id)
-		} else if vol.PersistentVolumeClaim != nil {
-			pvcName := vol.PersistentVolumeClaim.ClaimName
-			if pvcName == "" {
-				return fmt.Errorf("PersistentVolumeClaim had no name")
-			}
+			continue
+		}
 
+		pvcName := ""
+		ephemeral := false
+		switch {
+		case vol.PersistentVolumeClaim != nil:
+			pvcName = vol.PersistentVolumeClaim.ClaimName
+		case vol.Ephemeral != nil:
+			if !utilfeature.DefaultFeatureGate.Enabled(features.GenericEphemeralVolume) {
+				return fmt.Errorf(
+					"volume %s is a generic ephemeral volume, but that feature is disabled in kube-scheduler",
+					vol.Name,
+				)
+			}
+			// Generic ephemeral inline volumes also use a PVC,
+			// just with a computed name, and...
+			pvcName = pod.Name + "-" + vol.Name
+			ephemeral = true
+		default:
+			continue
+		}
+		if pvcName == "" {
+			return fmt.Errorf("PersistentVolumeClaim had no name")
```

Comment:
is it possible to create a pod with a pvc with no name?

Reply:
That if check was there already. I don't know whether this can happen in practice or whether it would be caught by validation. I prefer defensive programming, so I think it makes sense to have it.